src/index.js

```diff
@@ -107,24 +107,18 @@ export default function promiseMiddleware(config = {}) {
        *   }
        * }
        */
-      return new Promise((resolve, reject) => {
-        promise.then(
-          (value = null) => {
-            const resolvedAction = getAction(value, false);
-            dispatch(resolvedAction);
-            resolve({ value, action: resolvedAction });
-
-            return;
-          },
-          (reason = null) => {
-            const rejectedAction = getAction(reason, true);
-            dispatch(rejectedAction);
-            reject({ reason, action: rejectedAction });
-
-            return;
-          }
-        );
-      });
+      return promise
+        .then((value = null) => {
+          const resolvedAction = getAction(value, false);
+          dispatch(resolvedAction);
+          return { value, action: resolvedAction };
+        })
+        .catch((reason = null) => {
```

Comment:
Right, but you don't want the said dispatch errors to be swallowed either. So another catch is still useful, but shouldn't lead to a dispatch.